src/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -246,26 +246,26 @@ private synchronized static void synchronizedClear() {
   private void refreshGameList() {
     m_gameList.setEnabled(false);
     final NewGameChooserEntry selected = getSelected();
-    SwingUtilities.invokeLater(new Runnable() {
-      @Override
-      public void run() {
         try {
           refreshNewGameChooserModel();
           m_gameListModel = getNewGameChooserModel();
           m_gameList.setModel(m_gameListModel);
-          if (selected != null) {
-            final String name = selected.getGameData().getGameName();
-            final NewGameChooserEntry found = m_gameListModel.findByName(name);
-            if (name != null) {
-              m_gameList.setSelectedValue(found, true);
-            }
+          if(selected != null){
+            selectGame(selected.getGameData().getGameName());
           }
         } finally {
           m_gameList.setEnabled(true);
         }
+  }
+  
+  private void refreshGameListBackground(){
+    SwingUtilities.invokeLater(new Runnable() {
```

Comment:
FYI, a lambda can clean up the anonymous inner class:
` SwingUtilities.invokeLater( () -> refreshGameList() );`